### PR TITLE
added secret watching

### PR DESCRIPTION
**Pull request by xh3b4sd**
Towards https://github.com/giantswarm/operatorkit/issues/25.

### CODE REVIEW

File: cert.go [hunk 1]
```diff
@@ -0,0 +1,144 @@
...
+
+	return assetsBundle, nil
+}
+
+// SearchCertsForComponent watches for secrets of a single cluster component and
+// returns it as assets bundle.
+func (s *Service) SearchCertsForComponent(clusterID, componentName string) (AssetsBundle, error) {
+	// TODO we should also do a list. In case the secrets have already been
+	// created we might miss them with only watching.
+	watcher, err := s.k8sClient.Core().Secrets(api.NamespaceDefault).Watch(apismetav1.ListOptions{
```
Comment (kopiczko) [inline cert.go:90]: We should switch to informer like in kubeadmtokentpr.

**Review by kopiczko — COMMENTED** [review]
<3

File: cert.go (hunk 1 above)
Comment (xh3b4sd) [reply cert.go:90]: Agreed. This was just outside the scope of the original issue.

Comment (xh3b4sd) [reply cert.go:90]: Can I merge this though?

Comment (kopiczko) [reply cert.go:90]: Agreed, totally out of scope.

**Pull request merged**